src/Features/Core/Portable/CodeFixes/Suppression/WrapperCodeFixProvider.cs

```diff
@@ -12,10 +12,10 @@ internal sealed class WrapperCodeFixProvider : CodeFixProvider
         private readonly ImmutableArray<string> _originalDiagnosticIds;
         private readonly ISuppressionFixProvider _suppressionFixProvider;
 
-        public WrapperCodeFixProvider(ISuppressionFixProvider suppressionFixProvider, ImmutableArray<Diagnostic> originalDiagnostics)
+        public WrapperCodeFixProvider(ISuppressionFixProvider suppressionFixProvider, ImmutableArray<string> diagnosticIds)
```

Comment:
that means, again caller who has Diagnostics only need to convert them to DiagnosticData and then get ID and then through them away. the API only need ID so it should accept just id. we can create helper to get those if you think Select(..) is duplication.